src/main/java/seedu/address/model/recipe/Recipe.java

```diff
@@ -33,7 +33,7 @@
      * Every field must be present a not null, otherwise it throws a NullPointerException.
      */
     public Recipe(Name name, List<Ingredient> ingredients, CompletionTime completionTime,
-                  double portion, List<String> steps, Set<Tag> tags) {
+                  Portion portion, List<Step> steps, Set<Tag> tags) {
 
         //TODO: Parser needs to ensure that ingredients and steps are in a list.
         requireAllNonNull(name, ingredients, portion, steps, tags);
```

Comment:
As mentioned by @Oddcorner:
```suggestion
        requireAllNonNull(name, ingredients, completionTime, portion, steps, tags);
```